### PR TITLE
[SPARK-33680][SQL][TESTS] Fix PrunePartitionSuiteBase/BucketedReadWithHiveSupportSuite not to depend on the default conf

### PR DESCRIPTION
**Pull request by dongjoon-hyun**
### What changes were proposed in this pull request?

This PR updates `PrunePartitionSuiteBase/BucketedReadWithHiveSupportSuite` to have the require conf explicitly.

### Why are the changes needed?

The unit test should not depend on the default configurations.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

According to https://github.com/apache/spark/pull/30628 , this seems to be the only ones.

Pass the CIs.

### CODE REVIEW

**Comment by dongjoon-hyun** [comment]
Could you review this, @maropu and @viirya ?

File: sql/hive/src/test/scala/org/apache/spark/sql/sources/BucketedReadWithHiveSupportSuite.scala [hunk 1]
```diff
@@ -18,11 +18,18 @@
 package org.apache.spark.sql.sources
 
 import org.apache.spark.sql.hive.test.TestHiveSingleton
+import org.apache.spark.sql.internal.SQLConf.ADAPTIVE_EXECUTION_ENABLED
 import org.apache.spark.sql.internal.StaticSQLConf.CATALOG_IMPLEMENTATION
 
 class BucketedReadWithHiveSupportSuite extends BucketedReadSuite with TestHiveSingleton {
   protected override def beforeAll(): Unit = {
     super.beforeAll()
+    spark.sessionState.conf.setConf(ADAPTIVE_EXECUTION_ENABLED, false)
```
Comment (maropu) [inline sql/hive/src/test/scala/org/apache/spark/sql/sources/BucketedReadWithHiveSupportSuite.scala:27]: nit: could we use `DisableAdaptiveExecutionSuite` instead?

**Review by maropu — APPROVED** [review]
Looks fine if the tests pass.

File: sql/hive/src/test/scala/org/apache/spark/sql/sources/BucketedReadWithHiveSupportSuite.scala (hunk 1 above)
Comment (dongjoon-hyun) [reply sql/hive/src/test/scala/org/apache/spark/sql/sources/BucketedReadWithHiveSupportSuite.scala:27]: Oh, sure! Thanks.

**Review by HyukjinKwon — APPROVED** [review]

**Review by viirya — APPROVED** [review]

**Comment by dongjoon-hyun** [comment]
Thank you, @maropu , @HyukjinKwon , @viirya .
The updated tests in `hive` module passed. Merged to master/3.1.

**Comment by HyukjinKwon** [comment]
cc @gengliangwang, @cloud-fan and @maryannxue FYI it's CNF and AQE

**Comment by HyukjinKwon** [comment]
Sure, nice!

**Comment by HyukjinKwon** [comment]
cc @gengliangwang, @cloud-fan and @maryannxue FYI it's CNF and AQE

**Pull request closed without merging**